services/src/atdd-staging/src/main/java/org/openkilda/atdd/staging/service/flowcalculator/FlowCalculatorImpl.java

```diff
@@ -0,0 +1,185 @@
+import org.springframework.stereotype.Service;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class FlowCalculatorImpl implements FlowCalculator {
+    private static final SimpleDateFormat sdf = new SimpleDateFormat("ddMMMHHmm");
+
+    @Autowired
+    private NorthboundService northboundService;
+
+    @Autowired
+    private TopologyDefinition topologyDefinition;
+
+    @Autowired
+    private TopologyEngineService topologyEngineService;
+
+    /**
+     * Creates a number of flows with given number of alternate paths and at least 1 a-switch ISL. a-switch ISL
+     * will later allow to bring such link down and verify the further flow behavior. <br>
+     * Note that unlike some other methods here, the flow(s) will already be created in the system.
+     *
+     * @param flowsAmount    amount of flows to create. Will throw assumption error if unable to find enough flows in
+     *                       given topology
+     * @param alternatePaths amount of alternate paths that should be available for the created flows
+     * @param bw             bandwidth for created flows
+     * @return map. Key: created flow. Value: list of a-switch isls for flow.
+     */
+    @Override
+    public Map<FlowPayload, List<TopologyDefinition.Isl>> createFlowsWithASwitch(int flowsAmount,
+                                                                                 int alternatePaths, int bw) {
+
+        final List<TopologyDefinition.Switch> switches = topologyDefinition.getActiveSwitches();
+        FlowSetBuilder flowSet = new FlowSetBuilder();
+        boolean foundEnoughFlows = false;
+        Map<FlowPayload, List<TopologyDefinition.Isl>> result = new HashMap<>();
+
+        for (TopologyDefinition.Switch srcSwitch : switches) {
+            for (TopologyDefinition.Switch dstSwitch : switches) {
+                if (srcSwitch.getDpId().compareTo(dstSwitch.getDpId()) >= 0) {
+                    continue;
+                }
+                // test only bi-directional flow
+                List<PathInfoData> forwardPaths = topologyEngineService
+                        .getPaths(srcSwitch.getDpId(), dstSwitch.getDpId());
+                List<PathInfoData> reversePaths = topologyEngineService
+                        .getPaths(dstSwitch.getDpId(), srcSwitch.getDpId());
+                boolean hasAlternatePath = forwardPaths.size() > alternatePaths && reversePaths.size() > alternatePaths;
+                if (hasAlternatePath) {
+                    //try creating flow to see the actual path being used
+                    String flowId = format("%s-%s-%s", srcSwitch.getName(), dstSwitch.getName(),
+                            sdf.format(new Date()));
+                    FlowPayload flow = flowSet.new FlowBuilder(flowId, srcSwitch, dstSwitch)
```

Comment:
Why not create a method in FlowSetBuilder? Something like "flowSet.buildFlowInUniqueVlan(flowId, srcSwitch, dstSwitch, bw)"